dlp/sample.rb

```diff
@@ -0,0 +1,120 @@
+# Copyright 2018 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def inspect_string project_id: nil, content: nil, max_findings: 0
+  # [START dlp_inspect_string]
+  # project_id   = "Your Google Cloud project ID"
+  # content       = "The text to inspect"
+  # max_findings = "Maximum number of findings to report per request (0 = server maximum)"
+
+  require "google/cloud/dlp"
+
+  dlp = Google::Cloud::Dlp.new
+  inspect_config = {
+    # The types of information to match
+    info_types: [{name: "PERSON_NAME"}, {name: "US_STATE"}],
+
+    # Only return results above a likelihood threshold (0 for all)
+    min_likelihood: :POSSIBLE,
+
+    # Limit the number of findings (0 for no limit)
+    limits: { max_findings_per_request: max_findings },
+
+    # Whether to include the matching string in the response
+    include_quote: true
+  }
+
+  # The item to inspect
+  item_to_inspect = { value: content }
+
+  # Run request
+  parent = "projects/#{project_id}"
+  response = dlp.inspect_content parent,
+    inspect_config: inspect_config,
+    item:           item_to_inspect
+
+  # Print the results
+  response.result.findings.each { |finding|
+    puts "Quote:      #{finding.quote}"
+    puts "Info type:  #{finding.info_type.name}"
+    puts "Likelihood: #{finding.likelihood}"
+  }.empty? and begin
+    puts "No findings"
+  end
+  # [END dlp_inspect_string]
+end
+
+def inspect_file project_id: nil, filename: nil, max_findings: 0
```

Comment:
I missed this during my first pass. I think `max_findings should be available as a CLI parameter.

Reply:
done